src/Microsoft.FeatureManagement/FeatureFilters/Cron/CronField.cs

```diff
@@ -0,0 +1,241 @@
+                {
+                    return false;
+                }
+
+                int value = cronField.TryGetNumber(segment);
+                if (cronField.IsValueValid(value))
+                {
+                    cronField._bits[cronField.ValueToIndex(value)] = true;
+                    continue;
+                }
+
+                if (segment.Contains("-") || segment.Contains("*")) // The segment might be a range.
+                {
+                    if (string.Equals(segment, "*"))
+                    {
+                        cronField.MatchAll = true;
+                    }
+
+                    string[] parts = segment.Split('/');
+                    if (parts.Length > 2) // multiple slashs
+                    {
+                        return false;
+                    }
+
+                    int step = parts.Length == 2 ? cronField.TryGetNumber(parts[1]) : 1;
+                    if (step <= 0) // invalid step value 
+                    {
+                        return false;
+                    }
+
+                    string range = parts[0];
+                    int first, last;
+                    if (string.Equals(range, "*")) // asterisk represents unrestricted range
+                    {
+                        (first, last) = (cronField._minValue, cronField._maxValue);
+                    }
+                    else // range should be defined by two numbers separated with a hyphen
+                    {
+                        string[] numbers = range.Split('-');
+                        if (numbers.Length != 2)
+                        {
+                            return false;
+                        }
+
+                        (first, last) = (cronField.TryGetNumber(numbers[0]), cronField.TryGetNumber(numbers[1]));
+
+                        if (!cronField.IsValueValid(first) || !cronField.IsValueValid(last))
+                        {
+                            return false;
+                        }
+
+                        if (cronField._kind == CronFieldKind.DayOfWeek && last == 0 && last != first) // Corner case for Sunday: both 0 and 7 can be interpreted to Sunday
+                        {
+                            last = 7; // Mon-Sun should be intepreted to 1-7 instead of 1-0
+                        }
+
+                        if (first > last)
+                        {
+                            return false;
+                        }
+                    }
+
+                    for (int num = first; num <= last; num += step)
+                    {
+                        cronField._bits[cronField.ValueToIndex(num)] = true;
+                    }
+                }
+                else // The segment is neither a range nor a valid number.
+                {
+                    return false;
+                }
+            }
+
+            result = cronField;
+            return true;
+        }
+
+        private int TryGetNumber(string str)
```

Comment:
A method `TryGetNumber` should be `private bool TryGetNumber(string value, out int number)`